linker/Mono.Linker/LinkContext.cs

```diff
@@ -147,7 +147,7 @@ public AssemblyDefinition Resolve (string name)
 				return assembly;
 			}
 
-			return Resolve (new AssemblyNameReference (name, new Version ()));
+			return Resolve (new AssemblyNameReference (name, new Version (0, 0)));
```

Comment:
Another API that was removed. [https://github.com/dotnet/coreclr/pull/6810](https://github.com/dotnet/coreclr/pull/6810) gives a clue as to how the APIs available to build against differ from the coreclr sources. I don't fully understand when and how these are used, but it looks like everything is compiled against the reference assemblies such as ([mscorlib.cs](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/ref/mscorlib.cs)), which was missing the parameterless constructor. I think these are taken from nuget packages during the build.